data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/config/ThresholdConfig.java

```diff
@@ -0,0 +1,66 @@
+package org.opensearch.dataprepper.plugins.sink.config;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.Size;
+
+/**
+ * The threshold config holds the different configurations for
+ * buffer restrictions, retransmission restrictions and timeout
+ * restrictions.
+ */
+public class ThresholdConfig {
+    public static final int DEFAULT_BATCH_SIZE = 100;
+    public static final int DEFAULT_EVENT_SIZE = 50;
+    public static final int DEFAULT_SIZE_OF_REQUEST = 524288;
+    public static final int DEFAULT_RETRY_COUNT = 5;
+    public static final int DEFAULT_LOG_SEND_INTERVAL_TIME = 60;
+    public static final int DEFAULT_BACKOFF_TIME = 5000;
+
+    @JsonProperty("batch_size")
+    @Size(min = 1, max = 10000, message = "batch_size amount should be between 1 to 10000")
+    private int batchSize = DEFAULT_BATCH_SIZE;
+
+    @JsonProperty("max_event_size")
+    @Size(min = 1, max = 256, message = "max_event_size amount should be between 1 to 256 kilobytes")
+    private int maxEventSize = DEFAULT_EVENT_SIZE;
+
+    @JsonProperty("max_request_size")
+    @Size(min = 1, max = 1048576, message = "max_batch_request_size amount should be between 1 and 1048576 bytes")
+    private int maxRequestSize = DEFAULT_SIZE_OF_REQUEST;
+
+    @JsonProperty("retry_count")
```

Comment:
How is this retry count different from AWS SDK retry?

Reply:
This parameter is also based on the S3-Sink retry parameter "max_retries" as it attempts to retry recoverable requests.